### DISCUSSION TITLE
any update code search by date and 100 pages ?

### DISCUSSION THREAD

**Discussion by cobavcova** [community/community#121777] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

any update code search by date and 100 pages ?

**Comment by look** [comment, marked as answer]
Hi @cobavcova. We have no plans to implement search by date at this time because our code search backend de-duplicates file content and Git does not store the modification time at the tree or blob level. This makes the feature extremely difficult to implement correctly. 

As I've mentioned in the past, the old code search date filtering was based on when the file was indexed, and the index was rebuild infrequently. That made it a decent proxy for "last modified". The new system rebuilds the index frequently, so using the reindex time is not a useful way to sort.

We are planning to increase the result limit (we already did for the API) but it has been lower priority  than other work scaling the system.

**Reply by zb3** [reply]
>  our code search backend de-duplicates file content and Git does not store the modification time at the tree or blob level

So what about the time when this exact content was first seen (physically added to the db)? I'm aware that this is not semantically equivalent, but I'd even argue that that is what the user actually wants..

**Reply by look** [reply]
@zb3 that is a good idea, and we have considered it. However, it could leak information between Git repositories that share the same blobs, so we rejected that solution. Also, as I mentioned, we rebuild the index regularly, which would reset this timestamp, which would make it change. This would produce extremely confusing results.

**Reply by zb3** [reply]
Ouch, would that leak be problematic if only public code was indexed?
And you rebuild the index from scratch every time? Oh, seems counterintuitive, but at GitHub scale, many things are :D

**Reply by sgf** [reply]
I really can't understand Microsoft's behavior. You don't reduce costs through technology, but rely on castration of functions and degrading the experience. This is really puzzling.

**Reply by cachius** [reply]
@look If search by date doesn't work, please remove the special UI for the `sort:` keyword where the following text is highlighted in blue like a special search operator: 
<img width="159" height="33" alt="image" src="https://github.com/user-attachments/assets/c61f234a-d8a3-4d74-899a-1dc883874807" /> ([via](https://github.com/orgs/community/discussions/172625))

**Comment by winterdl** [comment]
The current code search feature is useless because other languages change a lot over time, but the search results cannot be sorted by time, so it only shows old results, making it not useful for reference.
